docs/source/posts/2024/2024-08-21-wachiou-final-report.rst

```diff
@@ -0,0 +1,282 @@
+The decorator works by inspecting the signature of the decorated function using the inspect module's signature function. It identifies function arguments that must be passed as named arguments (`KEYWORD_ONLY_ARGS`) and positional arguments (`POSITIONAL_ARGS`).
+When a decorated function is called, the decorator compares the arguments supplied with those expected by the signature. If mandatory named arguments are missing, or if positional arguments are used where named arguments are expected, the decorator attempts to correct these errors by transforming the remaining positional arguments into named arguments.
+In addition, the decorator issues a warning (`UserWarning`) if the current version of FURY is within a certain range (`from_version` to `until_version`). This warning informs the user that the current function call method will no longer be supported in future versions of FURY. If the FURY version exceeds until_version, an error is raised to indicate that the calling method is obsolete.
+
+This decorator was applied to several key FURY modules, including ``actors``, ``ui``, ``animation``, ``shares``, ``data``, and ``stream``. This enhanced the robustness and maintainability of the code, minimizing potential errors due to incorrect use of arguments.
+
+code source: https://github.com/fury-gl/fury/blob/master/fury/decorators.py
+
+*Pull Requests*:
+
+- Applied the ``warn_on_args_to_kwargs`` decorator to multiple modules:
+
+  - actors: https://github.com/fury-gl/fury/pull/898 (merged)
+  - animation: https://github.com/fury-gl/fury/pull/899 (merged)
+  - data: https://github.com/fury-gl/fury/pull/900 (merged)
+  - shares: https://github.com/fury-gl/fury/pull/901 (merged)
+  - ui: https://github.com/fury-gl/fury/pull/902 (merged)
+  - stream: https://github.com/fury-gl/fury/pull/909 (merged)
+
+- Initial implementation of the decorator: https://github.com/fury-gl/fury/pull/888 (merged)
+
+2. **Implement Lazy loading in FURY**
+
+Overview:
+The implementation of lazy loading in FURY was a key enhancement aimed at optimizing performance by delaying the loading of modules until they are actually needed. This approach reduces startup time and memory usage by only loading modules when they are accessed.
+
+- Implementation of lazy loading: Lazy loading was applied to various FURY modules using the lazy_loader module. This technique ensures that modules are only loaded into memory when their functionalities are first accessed, rather than during the initial import of the package.
+- Reading `SPEC1 <https://scientific-python.org/specs/spec-0001/>`: To ensure best practices, I reviewed the `SPEC1 <https://scientific-python.org/specs/spec-0001/>` document from Scientific Python. This document offered valuable insights and guidelines on implementing lazy loading effectively.
```

Comment:
this is strange, to fix

<img width="684" alt="image" src="https://github.com/user-attachments/assets/5618f798-4faf-41ce-819f-e5dd830ccdac">

Reply:
sure